vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java

```diff
@@ -231,6 +235,65 @@ public void findBlobs(StringComparisonRule rule, String blobNameToMatch, String
+     *                         <i>Available scopes:</i>
+     *                         <ul>
+     *                         <li><b>STEP</b> - the variable will be available only within the step,
+     *                         <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                         <li><b>STORY</b> - the variable will be available within the whole story,
+     *                         <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                         </ul>
+     * @param variableName     The variable name to store the list of found blob names.
+     * @param searchParameters Parameters to search blob names.
+     *                         <div>Example:</div>
+     *                         <code>
+     *                         <br>When I find blobs and save result to SCENARIO variable `blobs` with parameters::
+     *                         <br>|rule    |blobNameToMatch|container|storageAccountKey|directory|maxNumber|
+     *                         <br>|contains|file-key.txt   |global   |myStorage        |data/    |100      |
+     *                         </code>
+     *                         <br>
+     *                         <br>where
+     *                         <ul>
+     *                           <li><code>rule</code> The blob name comparison rule: "is equal to", "contains",
+     *                           "does not contain" or "matches".</li>
+     *                           <li><code>blobNameToMatch</code> The full or partial blob name to be matched.</li>
+     *                           <li><code>container</code> The name of the container to point to.</li>
+     *                           <li><code>storageAccountKey</code> The key to Storage Account endpoint.</li>
+     *                           <li><code>directory</code> The directory to list blobs underneath.</li>
+     *                           <li><code>maxNumber</code> Optional. Maximum number of blobs per page.</li>
+     *                          </ul>
+     */
+    @When("I find blobs and save result to $scopes variable `$variableName` with parameters:$searchParameters")
+    public void findBlobsByParameters(Set<VariableScope> scopes, String variableName, ExamplesTable searchParameters)
+    {
+        List<String> blobNames = new ArrayList<>();
+        for (Parameters row : searchParameters.getRowsAsParameters(true))
+        {
+            StringComparisonRule rule = row.valueAs("rule", StringComparisonRule.class);
+            String blobNameToMatch = row.valueAs("blobNameToMatch", String.class);
+            String container = row.valueAs("container", String.class);
+            String storageAccountKey = row.valueAs("storageAccountKey", String.class);
+            String directory = row.valueAs("directory", String.class);
+            ListBlobsOptions options = new ListBlobsOptions().setPrefix(directory);
```

Comment:
it seems it's not a true `directory`, but just prefix (I assume if there is a virtual directory: `/abc/def`, then it will work for both: `/abc` and `abc/d`)